### DISCUSSION TITLE
Question: Can I build without .mjs files?

### DISCUSSION THREAD

**Discussion by TwisterMc** [mozilla/pdf.js#18235] [Q&A] [answered]
I have a WordPress plugin that I maintain and it's a wrapper for PDF.js. However, I need `.js` files. `.mjs` files work as long as users update their web servers to support `.mjs` files; which is asking a lot for users who don't know anything about web development.

Is there a way I can build PDF.js on my own to compile to `.js`? Any help on how to do this would be appreciated.

**Comment by vanquish83** [comment]
I'm one the same boat

**Comment by TwisterMc** [comment, marked as answer]
Solution: Rename all `.mjs` files to `.js` and then rename all references to those files as well.

I've done this successfully.

**Reply by vanquish83** [reply]
Unfortunately, it doesn't work for me. Doing so, I get a **export declarations may only appear at top level of a module** error message.

Maybe I'm missing something. Could you create a zip file with your version working and share it with me?

**Reply by i-salameh95** [reply]
the same happen with me, I have renamed all files to js instead of mjs, but it shows multiple errors: Cannot use 'import.meta' outside a module (at pdf.min.mjs:21:102011)

**Reply by lovelyelfpop** [reply]
> 我也遇到了同样的问题，我已将所有文件重命名为 js 而不是 mjs，但它显示了多个错误：无法在模块外使用“import.meta”（在 pdf.min.mjs:21:102011）

same problem

**Reply by lovelyelfpop** [reply]
I use rollup to convert mjs to js

**Reply by TwisterMc** [reply]
You can't just change the file names, you need to change the imports too in the files.

If you really need a version of the plugin with JS, you can probably grab it here: https://plugins.trac.wordpress.org/browser/pdfjs-viewer-shortcode/tags/2.2.2/pdfjs
